src/SFA.DAS.Location.Infrastructure.UnitTests/ApiClient/WhenGettingDataFromPostcodeIO.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using SFA.DAS.Location.Domain.Configuration;
+using FluentAssertions;
+using System.Net;
+
+namespace SFA.DAS.Location.Infrastructure.UnitTests.ApiClient
+{
+    public class WhenGettingDataFromPostcodeIO
+    {
+        [Test, AutoData]
+        public async Task Then_The_Endpoint_Is_Called_And_Outcode_Data_Returned(
+            PostcodesLocationApiResponse postcodeResponse,
+            string query,
+            int count)
+        {
+            //Arrange
+            var response = new HttpResponseMessage
+            {
+                Content = new StringContent(JsonConvert.SerializeObject(postcodeResponse)),
+                StatusCode = System.Net.HttpStatusCode.Accepted
+            };
+            var httpMessageHandler = MessageHandler.SetupMessageHandlerMock(response, new Uri(string.Format(Constants.PostcodesUrl, query, count)));
+            var client = new HttpClient(httpMessageHandler.Object);
+            var postcodeService = new PostcodeApiService(client);
+
+            //Act
+            var actual = await postcodeService.GetAllStartingWithOutcode(query, count);
+
+            //Assert
+            actual.Should().BeEquivalentTo(postcodeResponse.Result);
+        }
+
+
+        [Test]
+        public void Then_If_It_Is_Not_Successful_An_Exception_Is_Thrown()
+
+        {
+            //Arrange
+            var response = new HttpResponseMessage
+            {
+
+                Content = new StringContent(""),
+                StatusCode = HttpStatusCode.BadRequest
+            };
+            var httpMessageHandler = MessageHandler.SetupMessageHandlerMock(response, new Uri(string.Format(Constants.PostcodesUrl, null, 10)));
+            var client = new HttpClient(httpMessageHandler.Object);
+            var postcodeService = new PostcodeApiService(client);
+
+            //Act Assert
+            Assert.ThrowsAsync<HttpRequestException>(() => postcodeService.GetAllStartingWithOutcode(null, 10));
+        }
+
+        [Test, AutoData]
+        public async Task Then_The_Endpoint_Is_Called_And_Postcode_Data_Returned(
+        PostcodeLocationApiResponse postcodeResponse,
+        string query)
+        {
+            var response = new HttpResponseMessage
+            {
+                Content = new StringContent(JsonConvert.SerializeObject(postcodeResponse)),
+                StatusCode = HttpStatusCode.Accepted,
+            };
+            var httpMessageHandler = MessageHandler.SetupMessageHandlerMock(response, new Uri(string.Format(Constants.PostcodeUrl, query)));
+            var client = new HttpClient(httpMessageHandler.Object);
+            var postcodeService = new PostcodeApiService(client);
+
+            //Act
+            var actual = await postcodeService.GetPostcodeData(query);
+
+            //Assert
+            actual.Should().BeEquivalentTo(postcodeResponse.Result);
+        }
+    }
```

Comment:
Missing a test case for the notfound response and returning null - I'd also split out the two methods under test into separate test files